docs/architecture/adr-002-evm-hooks.md

```diff
@@ -0,0 +1,52 @@
+# ADR 002: EVM Hooks
+
+## Changelog
+
+- 2021-08-11: first draft
+
+## Status
+
+PROPOSED
+
+## Abstract
+
+The current ADR proposes a hook interface to EVM module, with the goal of extend the functionality externally,
+specifically to support communication from EVM contract to native modules through logs.
+
+## Context
```

Comment:
This requires a more descriptive paragraph of the problem and the limitations of the current implementation.

> This section describes the forces at play, including technological, political, social, and project local. These forces are probably in tension, and should be called out as such. The language in this section is value-neutral. It is simply describing facts. It should clearly explain the problem and motivation that the proposal aims to resolve.

Reply:
Added some explanation for the motives to add hook interface.